File: generator/src/test/resources/kotlin/expected/jaxrs/RequestHeaderParamsTest/test-optional-header-parameter-generation.output.kt
package io.test.service

import io.test.Test
import javax.ws.rs.Consumes
import javax.ws.rs.DefaultValue
import javax.ws.rs.GET
import javax.ws.rs.HeaderParam
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.core.Response
import kotlin.Int
import kotlin.String

@Produces(value = ["application/json"])
@Consumes(value = ["application/json"])
public interface API {
  @GET
  @Path(value = "/tests")
  public fun fetchTest(
    @HeaderParam(value = "obj") obj: Test?,
    @HeaderParam(value = "str") str: String?,
    @HeaderParam(value = "int") int: Int?,
    @HeaderParam(value = "def1") @DefaultValue(value = "test") def1: String,
    @HeaderParam(value = "def2") @DefaultValue(value = "10") def2: Int,
  ): Response
}

File: generator/src/test/resources/swift/expected/RequestHeaderParamsTest/test-optional-header-parameter-generation.output.swift
import Sunday

public final class API<TransportType : Transport> : Sendable {

  public static var problemTypes: [ProblemRegistration] {
    return []
  }
  public let transport: TransportType
  public let defaultContentTypes: [MediaType]
  public let defaultAcceptTypes: [MediaType]

  public init(
    transport: TransportType,
    defaultContentTypes: [MediaType] = [],
    defaultAcceptTypes: [MediaType] = [.json],
    problemTypes: [ProblemRegistration] = API.problemTypes
  ) {
    self.transport = transport
    self.defaultContentTypes = defaultContentTypes
    self.defaultAcceptTypes = defaultAcceptTypes
    problemTypes.forEach { $0.register(on: transport) }
  }

  public func fetchTest(
    obj: Test? = nil,
    str: String? = nil,
    int: Int? = nil,
    def1: String? = "test",
    def2: Int? = 10
  ) throws -> Sunday.Operation<Empty, Test, TransportType> {
    return Sunday.Operation(
      transport: self.transport,
      spec: Sunday.OperationSpec(
        method: .get,
        pathTemplate: "/tests",
        pathParameters: nil,
        queryParameters: nil,
        body: Empty.none,
        contentTypes: nil,
        acceptTypes: self.defaultAcceptTypes,
        headers: [
          "obj": try ParameterValues.encode(obj),
          "str": try ParameterValues.encode(str),
          "int": try ParameterValues.encode(int),
          "def1": try ParameterValues.encode(def1),
          "def2": try ParameterValues.encode(def2)
        ].filter { $0.value != nil }
      )
    )
  }

}
